### PR TITLE
Refactor imagenet example.

### PR DESCRIPTION
**Pull request by mohitreddy1996**
- Remove no-op assignment of base learning rate in training code.
- In input pipeline, refactor to have repeat the tf dataset for both train and eval. 
  - Currently repeating of tf dataset is done for both, but is implemented as part of two different conditions.

### CODE REVIEW

**Comment by avital** [comment]
The PR LGTM but I'd like us to manually verify that the results haven't changed. Can you start a run before the change and after (either on Cloud or on Google infrastructure) and share links to Tensorboard.dev?

(Hopefully soon we'll have full automated end-to-end testing so I won't have to ask to do this manually :D )

**Comment by avital** [comment]
Or I guess, just curious if you're run this change and verified that it doesn't change behavior. You don't need to link to Tensorboard.dev if you're tested it.

**Comment by marcvanzee** [comment]
Assigned this to Mohit, since Avital asked him to verify that this change doesn't change behavior.

**Comment by mohitreddy1996** [comment]
Done as part of #287
